dhis-2/dhis-api/src/main/java/org/hisp/dhis/expressiondimensionitem/ExpressionDimensionItemHelper.java

```diff
@@ -0,0 +1,84 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.expressiondimensionitem;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.hisp.dhis.common.BaseDimensionalItemObject;
+import org.hisp.dhis.common.DataDimensionItem;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.indicator.Indicator;
+import org.hisp.dhis.program.ProgramDataElementDimensionItem;
+
+/**
+ * Parsing the expression of ExpressionDimensionItem, provides collection of
+ * BaseDimensionalItemObjects.
+ */
+public class ExpressionDimensionItemHelper
+{
+    private static final Pattern pattern = Pattern.compile( "[a-zA-Z0-9]{11}" );
+
+    private ExpressionDimensionItemHelper()
+    {
+        throw new UnsupportedOperationException( "helper" );
+    }
+
+    /**
+     * Provides collection of selected item types inside the expression
+     *
+     * @param manager {@link IdentifiableObjectManager} service for item
+     *        delivery
+     * @param dataDimensionItem {@link IdentifiableObjectManager} expression
+     *        dimension item
+     * @return collection of selected item types
+     */
+    public static List<BaseDimensionalItemObject> getExpressionItems( IdentifiableObjectManager manager,
+        DataDimensionItem dataDimensionItem )
+    {
+        if ( dataDimensionItem.getExpressionDimensionItem() == null )
+        {
+            return new ArrayList<>();
+        }
+
+        String expression = dataDimensionItem.getExpressionDimensionItem().getExpression();
+
+        List<String> expressionTokens = new ArrayList<>();
+
+        pattern.matcher( expression )
+            .results()
+            .map( mr -> mr.group( 0 ) )
+            .forEach( expressionTokens::add );
```

Comment:
Could we factor these lines out into a separate method, and add a unit test for that part only?